### PR TITLE
Add Datadog Terraform Module for AWS Lambda

### PR DESCRIPTION
**Pull request by duncanpharvey**
### What does this PR do?

Adds the Datadog Terraform Module for AWS Lambda and examples for how to use it.

### Motivation

Make it easier to instrument Datadog in AWS Lambda with Terraform

https://datadoghq.atlassian.net/browse/SVLS-4426

### Describe how to test/QA your changes

Follow the instructions using one of the examples to deploy a Lambda function instrumented with Datadog to AWS.

### CODE REVIEW

File: README.md
Comment (duncanista) [inline README.md:1]: Great work on this file

File: main.tf [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+  runtime_base_environment_variable_map = {
+    nodejs = {
+      DD_LAMBDA_HANDLER = var.handler
+    }
+    python = {
+      DD_LAMBDA_HANDLER = var.handler
+    }
+  }
+  runtime_base_handler_map = {
+    nodejs = "/opt/nodejs/node_modules/datadog-lambda-js/handler.handler"
```
Comment (duncanista) [inline main.tf:18]: We might have an issue when the user is not installing the layer, but the package, where the DD handler is different here, maybe we can change that in a future PR.

File: main.tf [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+      DD_SITE                    = "datadoghq.com"
+      DD_TRACE_ENABLED           = "true"
+    }
+    runtime = lookup(local.runtime_base_environment_variable_map, local.runtime_base, {})
+  }
+  handler = lookup(local.runtime_base_handler_map, local.runtime_base, var.handler)
+  layers = {
+    extension = ["${local.layer_name_base}:Datadog-Extension${local.layer_suffix}:${var.datadog_extension_layer_version}"]
+    lambda    = local.layer_runtime == null ? [] : ["${local.layer_name_base}:${local.layer_runtime}${local.runtime_base == "nodejs" ? "" : local.layer_suffix}:${local.datadog_lambda_layer_version}"]
+  }
```
Comment (duncanista) [inline main.tf:55]: Isn't there a better way to calculate this value? This doesn't look very readable.

**Review by duncanista — APPROVED** [review]
LGTM – left some nits!

File: main.tf (hunk 2 above)
Comment (duncanpharvey) [reply main.tf]: I broke out some of this business logic to separate variables to help with the readability

File: main.tf (hunk 1 above)
Comment (duncanpharvey) [reply main.tf:18]: Interesting, is that a common approach to not use the layers and instead install the package directly? We can gauge interest based on user feedback

Comment (duncanista) [reply main.tf:18]: Sometimes customers do it, definitely another PR, not now, until someone asks us to.

File: README.md
Comment (duncanpharvey) [reply README.md:1]: Thanks!

**Pull request merged**